storage/core/src/testFixtures/java/io/aiven/kafka/tieredstorage/storage/TestObjectKey.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2023 Aiven Oy
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.aiven.kafka.tieredstorage.storage;
+
+import java.util.Objects;
+
+public class TestObjectKey implements ObjectKey {
```

Comment:
Since we have this already, should we use it everywhere in tests instead of lambdas to avoid getting weird values from `toString()`?

Reply:
I changed in the most places, but decided to leave out everything in `:core`, as `TestObjectKey` is in `:storage:core` and I'm not sure it's worth trying to find a better place for it.